packages/core/src/interpreter.ts

```diff
@@ -131,158 +118,131 @@ export class Interpreter<
   public _parent?: ActorRef<any>;
   public name: string;
   public ref: ActorRef<TEvent>;
+  private _actorContext: ActorContext<TEvent, SnapshotFrom<TBehavior>>;
 
   /**
    * The globally unique process ID for this invocation.
    */
   public sessionId: string;
-  private forwardTo: Set<string> = new Set();
+
+  // TODO: remove
+  public _forwardTo: Set<string> = new Set();
 
   /**
    * Creates a new Interpreter instance (i.e., service) for the given machine with the provided options, if any.
    *
-   * @param machine The machine to be interpreted
+   * @param behavior The machine to be interpreted
    * @param options Interpreter options
    */
-  constructor(
-    public machine: StateMachine<
-      TContext,
-      TEvent,
-      any,
-      any,
-      TResolvedTypesMeta
-    >,
-    options?: InterpreterOptions
-  ) {
+  constructor(public behavior: TBehavior, options?: InterpreterOptions) {
     const resolvedOptions = {
       ...defaultOptions,
       ...options
     } as Required<InterpreterOptions>;
 
     const { clock, logger, parent, id } = resolvedOptions;
 
-    const resolvedId = id !== undefined ? id : machine.key;
-
-    this.name = this.id = resolvedId;
+    this.name = this.id = id;
```

Comment:
If they are the same then we should drop the other one and only use a single property.